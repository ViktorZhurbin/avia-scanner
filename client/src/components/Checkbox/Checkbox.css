.row {
    height: 36px;
    display: flex;
    align-items: center;
    position: relative;
    width: 100%;
    box-sizing: border-box;
    cursor: pointer;
    user-select: none;
    font: var(--p3);
    color: var(--color-twilight);
}

.input {
    display: none;
}

.checkmark {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 17px;
    width: 17px;
    border-radius: 3px;
    border: 1px solid var(--color-grey);
    background-color: var(--color-white);
    margin-right: 10px;
}

.input:checked ~ .checkmark {
    border-color: var(--color-blue);
    background-color: var(--color-metal-light-background);
}

.checkmark::after {
    display: none;
    content: "";
    width: 4px;
    height: 6px;
    border: solid var(--color-blue);
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
}

.input:checked ~ .checkmark::after {
    display: block;
}
