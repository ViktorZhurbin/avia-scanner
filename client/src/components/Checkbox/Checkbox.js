import React from 'react';
import PropTypes from 'prop-types';
import cl from 'classnames/bind';

import styles from './Checkbox.css';

const cx = cl.bind(styles);

const Checkbox = ({
    id,
    name,
    checked,
    onChange,
}) => {
    const onToggleFilter = () => onChange(id);

    return (
        /* eslint-disable jsx-a11y/label-has-associated-control,jsx-a11y/label-has-for */
        <label className={cx('row')}>
            <input
                className={cx('input')}
                id={id}
                checked={checked}
                name={name}
                type="checkbox"
                onChange={onToggleFilter}
            />
            <span className={cx('checkmark')} />
            {name}
        </label>
        /* eslint-enable jsx-a11y/label-has-associated-control,jsx-a11y/label-has-for */
    );
};

Checkbox.propTypes = {
    id: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
    ]).isRequired,
    checked: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func,
};

Checkbox.defaultProps = {
    onChange: null,
};

export default React.memo(Checkbox);
