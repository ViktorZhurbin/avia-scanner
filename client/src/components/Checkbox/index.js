import React from 'react';
import PropTypes from 'prop-types';
import cl from 'classnames/bind';

import styles from './index.css';

const cx = cl.bind(styles);

class Checkbox extends React.PureComponent {
    static propTypes = {
        id: PropTypes.oneOfType([
            PropTypes.number,
            PropTypes.string,
        ]).isRequired,
        checked: PropTypes.bool.isRequired,
        name: PropTypes.string.isRequired,
        onChange: PropTypes.func,
        selectedStops: PropTypes.objectOf(
            PropTypes.bool,
        ).isRequired,
    };

    static defaultProps = {
        onChange: null,
    };

    onToggleFilter = () => {
        const { onChange, selectedStops, id } = this.props;

        const selectedFilters = {
            ...selectedStops,
            [id]: !selectedStops[id],
        };
        onChange(selectedFilters);
    }

    render() {
        const {
            id,
            checked,
            name,
        } = this.props;

        return (
            // eslint-disable-next-line jsx-a11y/label-has-associated-control,jsx-a11y/label-has-for
            <label className={cx('row')}>
                <input
                    className={cx('input')}
                    id={id}
                    checked={checked}
                    name={name}
                    type="checkbox"
                    onChange={this.onToggleFilter}
                />
                <span className={cx('checkmark')} />
                {name}
            </label>
        );
    }
}

export default Checkbox;
