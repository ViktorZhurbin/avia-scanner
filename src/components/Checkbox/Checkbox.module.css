.row {
    padding: 0 15px;
    display: flex;
    align-items: center;
    position: relative;
    cursor: pointer;
    user-select: none;
    font-size: 13px;
    line-height: 2.69;
    color: #4a4a4a;
}

.row:hover {
  background-color: #f1fcff;
}

.input {
    display: none;
}

.checkmark {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 19px;
    width: 19px;
    border-radius: 3px;
    border: 1px solid #d2d5d6;
    background-color: #fff;
    margin-right: 10px;
}

.input:checked ~ .checkmark {
    border-color: #2196f3;
    background-color: #f2fcff;
}

.checkmark:after {
    display: none;
}

.input:checked ~ .checkmark:after {
    display: block;
}

.checkmark:after {
    content: "";
    width: 4px;
    height: 6px;
    border: solid #3e9ce8;
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
}