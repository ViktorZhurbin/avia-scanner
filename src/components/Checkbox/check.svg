<svg width="19" height="19" viewBox="0 0 19 19" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <defs>
        <path id="a" d="M8.6 3.09h1.213v8.347H6.176v-1.212h2.425z"/>
    </defs>
    <use fill="#2196f3" transform="rotate(45 7.995 7.263)" xlink:href="#a"/>
</svg>