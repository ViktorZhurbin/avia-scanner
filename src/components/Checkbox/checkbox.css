.row {
    padding: 0 15px;
    display: flex;
    align-items: center;
    position: relative;
    cursor: pointer;
    user-select: none;
    font: var(--p3);
    line-height: 35px;
    color: var(--color-grey);

    &:hover {
        background-color: var(--color-bg-white);
    }
}

.input {
    display: none;
}

.checkmark {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 19px;
    width: 19px;
    border-radius: 3px;
    border: 1px solid var(--color-bg-grey);
    background-color: var(--color-white);
    margin-right: 10px;
}

.input:checked ~ .checkmark {
    border-color: var(--color-blue);
    background-color: var(--color-bg-white);
}

.checkmark::after {
    display: none;
    content: "";
    width: 4px;
    height: 6px;
    border: solid var(--color-blue);
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
}

.input:checked ~ .checkmark::after {
    display: block;
}
