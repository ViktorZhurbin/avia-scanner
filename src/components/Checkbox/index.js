import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import styles from './Checkbox.module.css';

const cx = classNames.bind(styles);

const Checkbox = (props) => {
    const {
        id,
        checked,
        name,
        onChange,
    } = props;

    return (
        <label className={styles.row}> {/*eslint-disable-line*/}
            <input
                className={cx('input')}
                id={id}
                checked={checked}
                name={name}
                type="checkbox"
                onChange={onChange}
            />
            <span className={cx('checkmark')} />
            {name}
        </label>
    );
};

Checkbox.propTypes = {
    id: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
    ]).isRequired,
    checked: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func,
};

Checkbox.defaultProps = {
    onChange: null,
};

export default Checkbox;
