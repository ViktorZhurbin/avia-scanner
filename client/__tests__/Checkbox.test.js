import React from 'react';
import renderer from 'react-test-renderer';

import Checkbox from '../src/components/Checkbox';

const props = {
    id: 0,
    checked: false,
    name: 'checkbox',
    selectedStops: { 0: true, 1: false },
};

describe('Render Checkbox', () => {
    it('render correctly with required and default props', () => {
        const component = renderer.create(
            <Checkbox {...props} />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
