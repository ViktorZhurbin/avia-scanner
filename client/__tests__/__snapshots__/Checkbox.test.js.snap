// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Checkbox render correctly with required and default props 1`] = `
<label
  className="row"
>
  <input
    checked={false}
    className="input"
    id={0}
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="checkmark"
  />
  checkbox
</label>
`;
